### ISSUE TITLE
Nested schema unsupported

### ISSUE THREAD

**Issue by mengqing** [michaelherold/interactor-contracts#11]
```
class Foobar
  include Interactor
  include Interactor::Contracts

  expects do
    required(:foo).schema do
      required(:bar).filled
    end
  end
end

Foobar.call(foo: {})
=> TypeError: no implicit conversion of Hash into Array
```

**Comment by raykov** [comment]
I also faced this issue when tried to validate an array:

```ruby
    class Foo
      include Interactor
      include Interactor::Contracts

      expects do
        required(:bar).each do
          required(:buzz).filled(:str?, min_size?: 10)
        end
      end

      on_breach do |breaches|
        context.fail!(message: breaches.to_h)
      end
    end

    Foo.({bar: [{buzz: ""}]})
```

```
    TypeError: no implicit conversion of Hash into Array
    from /usr/local/bundle/gems/interactor-contracts-0.1.0/lib/interactor/contracts/breach_set.rb:39:in `|'
```

**Comment by michaelherold** [comment]
This was fixed by #16.
